### DISCUSSION TITLE
Should LICENSE contain a copyright notice ?

### DISCUSSION THREAD

**Discussion by roelz-em** [apache/logging-log4net#171] [General]
Should the copyright notice be completed ?
Copyright [yyyy] [name of copyright owner]
https://github.com/apache/logging-log4net/blob/master/LICENSE#L189C3-L189C46

**Comment by FreeAndNil** [comment]
@roelz-em thanks for the hint. I've adjusted the placeholder.

**Reply by ppkarwasz** [reply]
I am not certain the the ASF is the copyright holder of our code.

Unless I am mistaken, individual contributors remain the copyright holders of the code, but they license it to the ASF.

**Reply by FreeAndNil** [reply]
I've copied the line from
https://github.com/apache/logging-log4j2/blob/2.x/LICENSE.txt#L190

**Reply by FreeAndNil** [reply]
@ppkarwasz see ICLA §2
https://www.apache.org/licenses/icla.pdf

2. Grant of Copyright License. Subject to the terms and conditions of
 this Agreement, You hereby grant to the Foundation and to
 recipients of software distributed by the Foundation a perpetual,
 worldwide, non-exclusive, no-charge, royalty-free, irrevocable
 copyright license to reproduce, prepare derivative works of,
 publicly display, publicly perform, sublicense, and distribute Your
 Contributions and such derivative works.

**Reply by FreeAndNil** [reply]
@ppkarwasz I think you are right. There were already some discussions about the topic.
https://lists.apache.org/thread/5pt17s291y68oy26mbt0yk3jjovx221m
Should I revert my changes? Should I delete the Copyright line?

**Reply by ppkarwasz** [reply]
I would copy/paste the contents of https://www.apache.org/licenses/LICENSE-2.0.txt
to the `LICENSE.txt` file.

The line:
```
   Copyright [yyyy] [name of copyright owner]
```
should IMHO stay as is. It is there to tell people how to apply the license to a generic project.

For ASF projects it is more complex:
```
// Licensed to the Apache Software Foundation (ASF) under one or more 
// contributor license agreements. See the NOTICE file distributed with
// this work for additional information regarding copyright ownership. 
// The ASF licenses this file to you under the Apache License, Version 2.0
// (the "License");
```
We license the code to the ASF and the ASF licenses it back to users.

**Reply by roelz-em** [reply]
Thanks for the detailed information.

**Comment by roelz-em** [comment]
I am referring to https://github.com/apache/logging-log4net/blob/master/LICENSE
The page shows "License and copyright notice" under "Conditions"
The license file is part of the log4net repository.
On usage of log4net, a copyright notice shall be included in the software using it.
The license file does not contain a log4net copyright notice.

**Reply by ppkarwasz** [reply]
If you are talking about the annex to the license, IMHO it should stay as is:

https://github.com/apache/logging-log4net/blob/40dd3883a716dec6e273704c5254d36726d7caf9/LICENSE#L178-L201

That is just a template, the copyright statements are in the NOTICE file: https://github.com/apache/logging-log4net/blob/master/NOTICE
This file is in line with the [ASF source header and copyright notice policy](https://www.apache.org/legal/src-headers.html#notice).

Note also that the contributors grant the ASF a _Copyright License_, I doubt that means the ASF becomes the _Copyright Owner_. That is why the headers of our products don't have a copyright notice and just mention that the rightful copyright owners gave the ASF a lot of rights and the ASF gives those rights to you. @grobmeier, did I miss something?

**Reply by roelz-em** [reply]
Thanks. I did not know about the NOTICE file and expected the copyright notice to be contained in the LICENSE file (The license is referenced in the NuGet package).

**Reply by FreeAndNil** [reply]
@roelz-em we include the copyright additionally in the log4net.nuspec file inside the nuget package:
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>log4net</id>
    <version>3.0.0</version>
    <title>Apache log4net</title>
    <authors>The Apache Software Foundation</authors>
    ...
    <license type="expression">Apache-2.0</license>
    <licenseUrl>https://licenses.nuget.org/Apache-2.0</licenseUrl>
    <copyright>Copyright 2004-2024 The Apache Software Foundation</copyright>
    ...
  </metadata>
</package>
```
